databroker/_core.py

```diff
@@ -2145,6 +2145,17 @@ def from_config(cls, config, auto_register=True):
         for spec, handler in config.get('handlers', {}).items():
             cls = load_cls(handler)
             db.reg.register_handler(spec, cls)
+        # if 'root_map' in config, set the root_map
+        if 'root_map' in config:
+            root_map = config['root_map']
+            try:
+                if len(root_map) == 2:
+                    # change to dict
+                    root_map = {root_map[0] : root_map[1]}
+                    db.reg.set_root_map(root_map)
+            except TypeError:
+                warnings.warn("root_map is not right type in config file"
```

Comment:
I think this should be a `TypeError` instead of a warning, but it is good to catch the plain `TypeError` and re-raise one with more detail, as you have done. Note that this needs a trailing space so it does not read "config fileMust".